main.rb

```diff
@@ -0,0 +1,148 @@
+require_relative 'app'
+
+class Main
+  def initialize
+    @app = App.new
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialisation: '
+    specialisation = gets.chomp
+    puts 'Teacher created successfully '
+    puts
+    @app.create_teacher(age, specialisation, name)
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parents permission ? [y/n]: '
+    permission = gets.chomp
+    puts 'Student created successfully '
+    puts
+    @app.create_student(name, age, permission)
+  end
+
+  def create_person
+    loop do
+      print 'Pleas choose an option: '
+      print ' -(1) to create a student.'
+      print ' -(2) to create a teacher.'
+      print '[Please enter the corresponding number]: '
+      option = gets.chomp
+
+      case option
+      when '1'
+        create_student
+        break
+      when '2'
+        create_teacher
+        break
+      else
+        puts 'Option unknown'
+      end
+    end
+  end
+
+  def create_book
+    print 'Book Title: '
+    title = gets.chomp
+    print 'Book Author: '
+    author = gets.chomp
+    @app.create_book(title, author)
+    puts 'Great! Book created successfully'
+    puts
+  end
+
+  def create_rental
+    puts 'Choose a book below by entering the corresponding number:'
+    list_books(with_id: true)
+    book = gets.chomp
+    puts 'Choose a person from the following list by number the corresponding'
+    list_persons(with_id: true)
+    person = gets.chomp
+    print 'Date: '
+    date = gets.chomp
+    puts 'Great! Rental created successfully'
+    puts
+    @app.create_rental(date, book.to_i, person.to_i)
+  end
+
+  def list_rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals: '
+    rentals = @app.list_rentals(id.to_i)
+    rentals&.each do |rental|
+      puts "Date: \"#{rental.date}\", Book: #{rental.person.title} by #{rental.person.author}"
+    end
+    puts
+  end
```

Comment:
- [ ] Please from the project requirement, all these methods (method to list all books, method to list all people, method to create a person either teacher or student, method to create a book, method to create a rental and method to list all rentals for a given person id.) should be in the `app.rb` file which serves as your console app entry-point, and this `main.rb` should only define the entry point by presenting the user with a list of options to perform, It lets users choose an option, if needed, ask for parameters for the option and have a way to quit the app like example code below. :relaxed: 
    
    ```ruby
    # main.rb
    def main
      app = App.new()
    
      response = nil
    
      # [OPTIONAL] You can learn  about the ruby lessons about basic sintax like loops: https://github.com/microverseinc/curriculum-ruby/blob/main/simple-ruby/lessons/basic_syntax.md  
      # *Present the user with a list of options to perform.
      puts "Welcome to School Library App!\n\n"
      while response != "7"
        puts "Please choose an option by enterin a number:"
        puts "1 - List all books"
        puts "2 - List all people"
        puts "3 - Create a person"
        puts "4 - Create a book"
        puts "5 - Create a rental"
        puts "6 - List all rentals for a given person id"
        puts "7 - Exit"
        # *Lets users choose an option.
        response = gets.chomp
    
        case response
        when "1"
          app.list_books()
        when "2"
          app.list_people()
        when "3"
          app.create_person()
        when "4"
          app.create_book()
        when "5"
          app.create_rental()
        when "6"
          app.list_rentals_for_person_id()
        when "7"
          # *Have a way to quit the app.
          puts "Thank you for using this app!"
        end
    
        puts "\n"
      end
    end
    
    # 'main' method should be invoked at the end of the main.rb file, hint in this lesson https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point
    main()
    ```